### PR TITLE
Compile time check to enforce that list implements the necessary 

### PR DESCRIPTION
**Pull request by zouyee**
**What this PR does / why we need it**:

1、Compile time check to enforce that list implements the necessary interface 
2、using explicit version

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```NONE

```

### CODE REVIEW

File: pkg/volume/cephfs/cephfs.go [hunk 1]
```diff
@@ -40,7 +40,9 @@ type cephfsPlugin struct {
 	host volume.VolumeHost
 }
 
+// Compile time check to enforce that list implements the necessary interface.
 var _ volume.VolumePlugin = &cephfsPlugin{}
+var _ volume.PersistentVolumePlugin = &cephfsPlugin{}
```
Comment (resouer) [inline pkg/volume/cephfs/cephfs.go]: What's this change for?

File: pkg/controller/volume/attachdetach/attach_detach_controller_test.go [hunk 2]
```diff
@@ -155,7 +155,7 @@ func attachDetachRecoveryTestCase(t *testing.T, extraPods1 []*v1.Pod, extraPods2
 
 	stopCh := make(chan struct{})
 
-	pods, err := fakeKubeClient.Core().Pods(v1.NamespaceAll).List(metav1.ListOptions{})
+	pods, err := fakeKubeClient.CoreV1().Pods(v1.NamespaceAll).List(metav1.ListOptions{})
```
Comment (resouer) [inline pkg/controller/volume/attachdetach/attach_detach_controller_test.go]: Emm, are you expect performance improve from this change? (I am not quite sure if this fix make sense actually)

File: pkg/volume/cephfs/cephfs.go [hunk 3]
```diff
@@ -40,7 +40,9 @@ type cephfsPlugin struct {
 	host volume.VolumeHost
 }
 
+// Compile time check to enforce that list implements the necessary interface.
```
Comment (mikedanese) [inline pkg/volume/cephfs/cephfs.go]: list?

**Pull request closed without merging**